src/App.js

```diff
@@ -28,31 +27,106 @@ const generateSquares = () => {
 const App = () => {
 
   const [squares, setSquares] = useState(generateSquares());
-
+  const [currentPlayer, setCurrentPlayer] = useState(PLAYER_1)
+  const [winner, setCurrentWinner] = useState(null);
+  const [count, setCurrentCount] = useState(0)
   // Wave 2
   // You will need to create a method to change the square 
   //   When it is clicked on.
   //   Then pass it into the squares as a callback
 
+  const onClickCallback = (id) => {
+    if (winner === "PLAYER_1" || winner === "PLAYER_2") return
+    const newSquare = [...squares] 
+    let i = 0
+    while(i < squares.length){
+      let x = 0
+      while (x < squares.length){
+        if (squares[i][x].id === id && squares[i][x].value === "" ){
+          squares[i][x].value = currentPlayer? PLAYER_1 : PLAYER_2;
+          setCurrentPlayer(!currentPlayer)
+          setCurrentWinner(checkForWinner())
+          setCurrentCount(count + 1)
+        };    
+        x += 1
+      }
+      i += 1
+    }
+   setSquares(newSquare);
+  }
 
   const checkForWinner = () => {
     // Complete in Wave 3
 
+    const newSquare = [...squares]
+    const flatSquare = newSquare.flat()
+
+    if(flatSquare[0].value !== '' && 
+    (flatSquare[0].value === flatSquare[4].value && 
+      flatSquare[0].value === flatSquare[8].value) ) {
+      if(flatSquare[0].value === 'X') {
+        return "PLAYER_1"
+      } else {
+        return "PLAYER_2"
+      };
+    } else if (flatSquare[2].value !== '' && 
+    (flatSquare[2].value === flatSquare[4].value && 
+      flatSquare[2].value === flatSquare[6].value) ) {
+        if(flatSquare[2].value === 'X') {
+          return "PLAYER_1"
+        } else {
+          return "PLAYER_2"
+        };
+
+      } else {
+      let x = 0
+      let i = 0
+      while(x < flatSquare.length && i < flatSquare.length){
+        if (flatSquare[x].value !== "" &&
+          flatSquare[x].value === flatSquare[x + 3].value &&
+          flatSquare[x].value === flatSquare[x + 6].value){
```

Comment:
I'm just pointing out that you don't need to do this type of math if you keep the array 2 dimensional.